### PR TITLE
sync-react: Handle version bumps across SemVer minors and release channels

### PR DESCRIPTION
**Pull request by eps1lon**
Automated syncs are currently failing since we changed the SemVer minor and release channel in the version string (e.g. https://github.com/vercel/next.js/actions/runs/12399391057/job/34614192762). Now we should handle these cases gracefully.

## test plan
- [x] `pnpm sync-react --version "19.1.0-canary-7eb8234f-20241218"`
- [x] https://github.com/vercel/next.js/actions/runs/12400884176 -> https://github.com/vercel/next.js/pull/74095

### CODE REVIEW

**Comment by eps1lon** [comment]
* **#74091** <a href="https://app.graphite.dev/github/pr/vercel/next.js/74091?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/vercel/next.js/74091?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: scripts/sync-react.js [hunk 1]
```diff
@@ -292,24 +302,20 @@ Or, run this command with no arguments to use the most recently published versio
...
   if (commit) {
     await commitEverything('Update `react@experimental`')
   }
   await sync({
-    newDateString,
-    newSha,
     newVersionStr,
     noInstall: !install,
-    channel: 'rc',
+    channel: '<framework-stable>',
```
Comment (eps1lon) [inline scripts/sync-react.js:315]: There isn't a meaningful channel here. It's whatever you passed in `--version` which is what we consider "framework stable".

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| buildDuration | 19.6s | 15.8s | N/A |
| buildDurationCached | 15.1s | 13s | N/A |
| nodeModulesSize | 416 MB | 416 MB | ✓ |
| nextStartRea..uration (ms) | 481ms | 476ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| 1187-HASH.js gzip | 52.4 kB | 52.4 kB | N/A |
| 8276.HASH.js gzip | 169 B | 168 B | N/A |
| 8377-HASH.js gzip | 5.36 kB | 5.36 kB | N/A |
| bccd1874-HASH.js gzip | 53 kB | 53 kB | N/A |
| framework-HASH.js gzip | 57.5 kB | 57.5 kB | N/A |
| main-app-HASH.js gzip | 232 B | 235 B | N/A |
| main-HASH.js gzip | 34.1 kB | 34.1 kB | N/A |
| webpack-HASH.js gzip | 1.71 kB | 1.71 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 39.4 kB | 39.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| _app-HASH.js gzip | 193 B | 193 B | ✓ |
| _error-HASH.js gzip | 193 B | 193 B | ✓ |
| amp-HASH.js gzip | 512 B | 510 B | N/A |
| css-HASH.js gzip | 343 B | 342 B | N/A |
| dynamic-HASH.js gzip | 1.84 kB | 1.84 kB | ✓ |
| edge-ssr-HASH.js gzip | 265 B | 265 B | ✓ |
| head-HASH.js gzip | 363 B | 362 B | N/A |
| hooks-HASH.js gzip | 393 B | 392 B | N/A |
| image-HASH.js gzip | 4.49 kB | 4.49 kB | N/A |
| index-HASH.js gzip | 268 B | 268 B | ✓ |
| link-HASH.js gzip | 2.35 kB | 2.34 kB | N/A |
| routerDirect..HASH.js gzip | 328 B | 328 B | ✓ |
| script-HASH.js gzip | 397 B | 397 B | ✓ |
| withRouter-HASH.js gzip | 323 B | 326 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.59 kB | 3.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| _buildManifest.js gzip | 749 B | 746 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| index.html gzip | 523 B | 524 B | N/A |
| link.html gzip | 538 B | 538 B | ✓ |
| withRouter.html gzip | 519 B | 521 B | N/A |
| Overall change | 538 B | 538 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| edge-ssr.js gzip | 129 kB | 129 kB | N/A |
| page.js gzip | 205 kB | 205 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 668 B | 668 B | ✓ |
| middleware-r..fest.js gzip | 155 B | 156 B | N/A |
| middleware.js gzip | 31.3 kB | 31.3 kB | N/A |
| edge-runtime..pack.js gzip | 844 B | 844 B | ✓ |
| Overall change | 1.51 kB | 1.51 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| 274-experime...dev.js gzip | 322 B | 322 B | ✓ |
| 274.runtime.dev.js gzip | 314 B | 314 B | ✓ |
| app-page-exp...dev.js gzip | 356 kB | 356 kB | ✓ |
| app-page-exp..prod.js gzip | 129 kB | 129 kB | ✓ |
| app-page-tur..prod.js gzip | 142 kB | 142 kB | ✓ |
| app-page-tur..prod.js gzip | 137 kB | 137 kB | ✓ |
| app-page.run...dev.js gzip | 346 kB | 346 kB | ✓ |
| app-page.run..prod.js gzip | 125 kB | 125 kB | ✓ |
| app-route-ex...dev.js gzip | 37.5 kB | 37.5 kB | ✓ |
| app-route-ex..prod.js gzip | 25.5 kB | 25.5 kB | ✓ |
| app-route-tu..prod.js gzip | 25.5 kB | 25.5 kB | ✓ |
| app-route-tu..prod.js gzip | 25.3 kB | 25.3 kB | ✓ |
| app-route.ru...dev.js gzip | 39.1 kB | 39.1 kB | ✓ |
| app-route.ru..prod.js gzip | 25.3 kB | 25.3 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.69 kB | 9.69 kB | ✓ |
| pages-api.ru...dev.js gzip | 11.6 kB | 11.6 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.68 kB | 9.68 kB | ✓ |
| pages-turbo...prod.js gzip | 21.7 kB | 21.7 kB | ✓ |
| pages.runtim...dev.js gzip | 27.5 kB | 27.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.7 kB | 21.7 kB | ✓ |
| server.runti..prod.js gzip | 916 kB | 916 kB | ✓ |
| Overall change | 2.43 MB | 2.43 MB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/12-18-sync-react_handle_version_bumps_across_semver_minors_and_release_channels | Change |
| - | - | - | - |
| 0.pack gzip | 2.08 MB | 2.08 MB | ⚠️ +764 B |
| index.pack gzip | 73.6 kB | 74.3 kB | ⚠️ +795 B |
| Overall change | 2.15 MB | 2.15 MB | ⚠️ +1.56 kB |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: fd164720854e81a90f0f287c563b1861c9c81134

**Comment by ijjk** [comment]
## Failing test suites 
Commit: fd164720854e81a90f0f287c563b1861c9c81134

`__NEXT_EXPERIMENTAL_PPR=true pnpm test-dev test/e2e/persistent-caching/persistent-caching.test.ts` (PPR)
- persistent-caching > should persistent cache loaders

<details>
<summary>Expand output</summary>

  ● persistent-caching › should persistent cache loaders

    expect(received).toBe(expected) // Object.is equality

    Expected: "Timestamp = 1734552442155"
    Received: "Timestamp = 1734552463123"

      55 |       const browser = await next.browser('/pages')
      56 |       // TODO Persistent Caching for webpack dev server is broken
    > 57 |       expect(await browser.elementByCss('main').text()).toBe(pagesTimestamp)
         |                                                         ^
      58 |       await browser.close()
      59 |     }
      60 |   })

      at Object.toBe (e2e/persistent-caching/persistent-caching.test.ts:57:57)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by huozhi — APPROVED** [review]

**Pull request merged**